### PR TITLE
LPS-194914 KB Article - Save the display date

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
First commit is about some changes Brian asked on https://github.com/brianchandotcom/liferay-portal/pull/140450

Screenshot 2023-09-11 at 15 42 55
Screenshot 2023-09-11 at 15 43 39

### CODE REVIEW

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Review by jaimelr10 — CHANGES_REQUESTED** [review]
Do we need to do anything regarding the `new Date()` occurences in here https://github.com/liferay-lima/liferay-portal/pull/4545/files? :thinking:

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java [hunk 1]
```diff
@@ -92,13 +94,23 @@ protected void doTransactionalCommand(
...
 		Date displayDate = new Date();
+
+		if (Validator.isNotNull(sDisplayDate)) {
+			displayDate = new SimpleDateFormat(
+				"yyyy-MM-dd HH:mm"
+			).parse(
+				sDisplayDate
+			);
+		}
+
```
[the remark this answers is not in the document]
Comment (ambrinchaudhary) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java]: That works perfectly! Thank you for the tip ;-)

**Comment by sofiamendozalr** [comment]
LGTM :smile:

**Comment by ambrinchaudhary** [comment]
> Do we need to do anything regarding the `new Date()` occurences in here https://github.com/liferay-lima/liferay-portal/pull/4545/files? 🤔

I don't know... some of them are from headless or tests, and in others it takes the displayDate already... you guys better know if we should change it somewhere else

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java [hunk 2]
```diff
@@ -92,13 +94,15 @@ protected void doTransactionalCommand(
 		String content = ParamUtil.getString(actionRequest, "content");
 		String description = ParamUtil.getString(actionRequest, "description");
 		String sourceURL = ParamUtil.getString(actionRequest, "sourceURL");
+		Date displayDate = ParamUtil.getDate(
+			actionRequest, "displayDate",
+			new SimpleDateFormat("yyyy-MM-dd HH:mm"));
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java:99]: Can you use `DateFormatFactoryUtil` to create the date format? Look for other usages of ParamUtil.getDate.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java (hunk 2 above)
Comment (ambrinchaudhary) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java]: Done! See https://github.com/liferay-lima/liferay-portal/pull/4677/commits/7220fc8c6e660a140dca8e9d5fb7d56a9b7a29b6

**Review by adolfopa — APPROVED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java [hunk 3]
```diff
@@ -92,13 +93,16 @@ protected void doTransactionalCommand(
 		String content = ParamUtil.getString(actionRequest, "content");
 		String description = ParamUtil.getString(actionRequest, "description");
 		String sourceURL = ParamUtil.getString(actionRequest, "sourceURL");
+		Date displayDate = ParamUtil.getDate(
+			actionRequest, "displayDate",
+			DateFormatFactoryUtil.getSimpleDateFormat("yyyy-MM-dd HH:mm"),
+			new Date());
```
Comment (jaimelr10) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/UpdateKBArticleMVCActionCommand.java:99]: Is this last paran `new Date()`  necessary? 
The desciption of  `ParamUtil.getDate()` without that last parameter says:

> Returns the portlet request parameter value as a Date. **If the parameter is missing or not convertible to a Date, the current date is returned**

**Review by jaimelr10 — CHANGES_REQUESTED** [review]

**Comment by ambrinchaudhary** [comment]
With 2 approvals and as the [last change](https://github.com/liferay-lima/liferay-portal/pull/4677#discussion_r1324050869) requested by Jaime is done, I'm going to take advantage and forward this.

**Comment by ambrinchaudhary** [comment]
ci:forward

**Review by jaimelr10 — APPROVED** [review]
LGTM!

**Comment by WYZ-Issac** [comment]
Failures are unrelated. Forward directly.

**Pull request closed without merging**